proto/commonfate/access/v1alpha1/access.proto

```diff
@@ -115,14 +115,18 @@ message BatchEnsureResponse {
   string trace_id = 1;
 
 
-  repeated GrantState grants = 2;
  
   // User-facing diagnostic warnings, for messages such as the user not having permissions
   // to request some of the entitlements they asked for.
   //
   // Rather than failing the entire operation the Access Handler returns warnings.
   repeated Diagnostic diagnostics = 6;
-  string request_id = 7;
+
+  repeated AccessRequest access_requests = 7;
+
+  repeated GrantState changes = 9;
```

Comment:
In order to make this a non breaking change, you need to keep the original number the same.

You can safely change the field name from grants to changes though as this only affects generated code.

It is technically a breaking change for users of the API but its only us for now so that is ok

```suggestion
  repeated GrantState changes = 2;
```